website/docs/r/api_gateway_rest_api.html.markdown

```diff
@@ -229,6 +229,7 @@ The following arguments are supported:
 * `endpoint_configuration` - (Optional) Configuration block defining API endpoint configuration including endpoint type. Defined below.
 * `minimum_compression_size` - (Optional) Minimum response size to compress for the REST API. Integer between `-1` and `10485760` (10MB). Setting a value greater than `-1` will enable compression, `-1` disables compression (default). If importing an OpenAPI specification via the `body` argument, this corresponds to the [`x-amazon-apigateway-minimum-compression-size` extension](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-openapi-minimum-compression-size.html). If the argument value (_except_ `-1`) is provided and is different than the OpenAPI value, the argument value will override the OpenAPI value.
 * `name` - (Required) Name of the REST API. If importing an OpenAPI specification via the `body` argument, this corresponds to the `info.title` field. If the argument value is different than the OpenAPI value, the argument value will override the OpenAPI value.
+* `fail_on_warnings` - (Optional) Whether warnings while API Gateway is creating or updating the resource should return an error or not. Defaults to `false`
```

Comment:
This argument is only applicable when a body (OpenAPI definition) argument is supplied.

```suggestion
* `fail_on_warnings` - (Optional) Failure mode of the PutRestApi operation when importing an OpenAPI specification via the body argument (create or update operation). Indicates whether to rollback the API update (true) or not (false) when a warning is encountered. Defaults to `false`.
```